sys/include/net/gnrc/netif.h

```diff
@@ -252,6 +252,164 @@ gnrc_netif_t *gnrc_netif_iter(const gnrc_netif_t *prev);
+ *          corresponding solicited-nodes multicast address.
+ * @return  -ENOTSUP, if @p netif doesn't support IPv6.
+ */
+static inline int gnrc_netif_ipv6_addr_add(const gnrc_netif_t *netif,
+                                           ipv6_addr_t *addr, unsigned pfx_len,
+                                           uint8_t flags)
+{
+    assert(netif != NULL);
+    assert(addr != NULL);
+    assert((pfx_len > 0) && (pfx_len <= 128));
+    return gnrc_netapi_set(netif->pid, NETOPT_IPV6_ADDR,
+                           ((pfx_len << 8U) | flags), addr,
+                           sizeof(ipv6_addr_t));
+}
+
+/**
+ * @brief   Removes a (unicast or anycast) IPv6 address from an interface (if
+ *          IPv6 is supported)
+ *
+ * @pre `netif != NULL`
+ * @pre `addr != NULL`
+ *
+ * @param[in] netif     The interface. May not be `NULL`.
+ * @param[in] addr      The address to remove from @p netif. May not be `NULL`.
+ *
+ * @return  sizeof(ipv6_addr_t) on success.
+ * @return  -ENOTSUP, if @p netif doesn't support IPv6.
+ */
+static inline int gnrc_netif_ipv6_addr_remove(const gnrc_netif_t *netif,
+                                              ipv6_addr_t *addr)
+{
+    assert(netif != NULL);
+    assert(addr != NULL);
+    return gnrc_netapi_set(netif->pid, NETOPT_IPV6_ADDR_REMOVE,
+                           0, addr, sizeof(ipv6_addr_t));
+}
+
+/**
+ * @brief   Gets the IPv6 multicast groups an interface is joined to (if IPv6 is
+ *          supported)
+ *
+ * @pre `netif != NULL`
+ * @pre `groups != NULL`
+ * @pre `max_len >= sizeof(ipv6_addr_t)`
+ *
+ * @param[in] netif     The interface. May not be `NULL`.
+ * @param[out] groups   Up to the first `max_len / 8` multicast groups @p netif
+ *                      is joined to. Must not be `NULL`
+ * @param[in] max_len   Number of *bytes* available in @p groups. Must be at
+ *                      least `sizeof(ipv6_addr_t)`. It is recommended to use
+ *                      @p GNRC_NETIF_IPV6_GROUPS_NUMOF `* sizeof(ipv6_addr_t)
+ *                      here (and have @p groups of the according length).
+ *
+ * @return  Number of addresses in @p groups times `sizeof(ipv6_addr_t)` on
+ *          success (including 0).
+ * @return  -ENOTSUP, if @p netif doesn't support IPv6.
+ */
+static inline int gnrc_netif_ipv6_groups_get(const gnrc_netif_t *netif,
+                                             ipv6_addr_t *groups,
+                                             size_t max_len)
+{
+    assert(netif != NULL);
+    assert(groups != NULL);
+    assert(max_len >= sizeof(ipv6_addr_t));
+    return gnrc_netapi_get(netif->pid, NETOPT_IPV6_GROUP, 0, groups, max_len);
+}
+
+/**
+ * @brief   Joins an IPv6 multicast group on an interface (if IPv6 is supported)
+ *
+ * @pre `netif != NULL`
+ * @pre `group != NULL`
+ *
+ * @param[in] netif     The interface.
+ * @param[in] group     The address of the multicast group to join on @p netif.
```

Comment:
Maybe add comments here to clarify the restrictions as done with the ipv6_groups_get function

Reply:
(done in the line below)